pyqtgraph/graphicsItems/PlotDataItem.py

```diff
@@ -38,6 +38,85 @@ class PlotDataItem(GraphicsObject):
     sigClicked = QtCore.Signal(object, object)
     sigPointsClicked = QtCore.Signal(object, object, object)
     sigPointsHovered = QtCore.Signal(object, object, object)
+    
+    class PlotDataset(object):
```

Comment:
I think this might be the first place in the library we a re using nested classes.  These are much tougher to test, so unless there is a good reason to test them, we should probably break this out into its own separate class.  

Also, if there is a chance we might want this class to emit signals, we should inherit from QObject as well, but since I'm not seeing a signal being emitted, probably no need to modify that.